src/Illuminate/View/Component.php

```diff
@@ -81,6 +81,10 @@ protected function createBladeViewFromString($factory, $contents)
         );
 
         if (! file_exists($viewFile = $directory.'/'.sha1($contents).'.blade.php')) {
+            if (! is_dir($directory)) {
```

Comment:
Should we check the OS here, and only do this IO if we are on windows. Otherwise, we are making things slower for everyone else for no reason? I think Laravel has a helper for checking if we are running on Windows?

Reply:
I think this operation only runs for one request and once the file is created the following requests do not hit this line, If I am not wrong.